File: packages/api-models-ts/src/Healpix.ts
/**
 * Request and response models for `/api/healpix`.
 *
 * @since 1.0.0
 */

import * as v from "valibot";

import * as Schemas from "./Schemas.ts";

/**
 * Counts of objects with and without a HEALPix index.
 *
 * @since 1.0.0
 * @category Models
 */
export const HealpixCounts = Schemas.model(
    v.strictObject({
        totalWithoutHealpix: v.optional(Schemas.Integer, 0),
        totalWithHealpix: v.optional(Schemas.Integer, 0),
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type HealpixCounts = v.InferOutput<typeof HealpixCounts>;

/**
 * Result of a HEALPix backfill batch.
 *
 * @since 1.0.0
 * @category Models
 */
export const HealpixUpdate = Schemas.model(
    v.strictObject({
        totalMatches: v.optional(Schemas.Integer, 0),
        pageNumber: v.optional(Schemas.Integer, 1),
        numPerPage: v.optional(Schemas.Integer, 100),
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type HealpixUpdate = v.InferOutput<typeof HealpixUpdate>;

File: packages/api-models-py/skyportal_py_models/healpix.py
"""Request and response models for ``/api/healpix``."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class HealpixCounts(BaseModel):
    """Counts of objects with and without a HEALPix index."""

    model_config = ConfigDict(extra="forbid", validate_by_name=True)

    total_without_healpix: int = Field(alias="totalWithoutHealpix", default=0)
    total_with_healpix: int = Field(alias="totalWithHealpix", default=0)


class HealpixUpdate(BaseModel):
    """Result of a HEALPix backfill batch."""

    model_config = ConfigDict(extra="forbid", validate_by_name=True)

    total_matches: int = Field(alias="totalMatches", default=0)
    page_number: int = Field(alias="pageNumber", default=1)
    num_per_page: int = Field(alias="numPerPage", default=100)
